opendatahub-io/modelmesh-serving
# restructured odh manifests

#### Motivation
Opendatahub manifests structure is different from upstream manifests. For this reason, it was difficult to synchronize upstream sources to ODH repos. Moreover, fvt test was not working with opendatahub manifests.
The change will help 
- to check if upstream manifests is the same as opendatahub manifests. 
- to generate a new manifests based on upstream manifests
- to deploy some required components for FVT test with opendatahub manifests.

#### Modifications
It includes 
- opendatahub folders
  -  scripts
  - scripts/manifessts
  - odh-manifests
- Makefile update


#### How to test?
1. Generate a new manifests and check if it is the same as the previous one
~~~
./opendatahub/scripts/gen_odh_modelmesh_manifests.sh
Generate opendatahub manifest in the /tmp/modelmesh-20230424041682356897
TARGET DIR: /tmp/modelmesh-20230424041682356897
--------------------------------------------------

 ✓ Creating a /tmp/modelmesh-20230424041682356897 folder
 ✓ Copying the model-mesh_templates to /tmp/modelmesh-20230424041682356897 folder
 ✓ Git Cloning odh-modelmesh-controller to /tmp/modelmesh-20230424041682356897 folder
 ✓ Copying the odh-modelmesh-controller manifests to model-mesh_templates folder
 ✓ Comment out ClusterServingRuntime
 ✓ Remove builtInServerTypes
 ✓ Add parameter variable into modelmesh-controller-rolebinding
 ✓ Replace each image of the parameter variable 'images'
 ✓ Replace replicas of odh modelmesh controller to 3
New Manifests are identical with previous one. You don't need to send any PR to ODH-MANIFESTS repo
~~~

2. Deploy opendatahub manifest
Refer ./opendatahub/docs/manifests-management.md
This doc explains how to deploy and how to test.

#### PR checklist

Checklist items below are applicable for development targeted to both fast and stable branches/tags
- [ ] FVT tests pass locally

Checklist items below are applicable for development targeted to both fast and stable branches/tags
- [ ] Tested modelmesh serving deployment with odh-manifests and ran odh-manifests-e2e tests locally 

==> This is odh-manifest changes so it does not impact these tests.

Issue: 
- https://issues.redhat.com/browse/RHODS-8011
- https://issues.redhat.com/browse/RHODS-8012
- https://issues.redhat.com/browse/RHODS-8013
